replication/row_event.go

```diff
@@ -1161,10 +1254,23 @@ func (e *RowsEvent) decodeValue(data []byte, tp byte, meta uint16) (v interface{
 		// Refer: https://github.com/shyiko/mysql-binlog-connector-java/blob/master/src/main/java/com/github/shyiko/mysql/binlog/event/deserialization/AbstractRowsEventDataDeserializer.java#L404
 		length = int(FixedLengthInt(data[0:meta]))
 		n = length + int(meta)
-		var d []byte
-		d, err = e.decodeJsonBinary(data[meta:n])
-		if err == nil {
-			v = hack.String(d)
+
+		if isPartial {
+			var diff *JsonDiff
+			diff, err = e.decodeJsonPartialBinary(data[meta:n])
+			if err == nil {
+				if diff == nil {
+					v = []byte{}
```

Comment:
Can we add a "JsonDiffOperationNoOp" JsonDiff, or tell the caller nil is no-op for JsonDiff? It's a bit ambiguous that we return a `[]byte{}`

Reply:
MySQL doesn't allow any other variants. So I just removed a nil JsonDiff :)